### ISSUE TITLE
Set Position Doesn't Work as Expected on iOS

### ISSUE THREAD

**Issue by mwpb** [inthehand/InTheHand.Forms#10]
Hi and thank you for making this useful package!

I'm having trouble using the setter for the Position with the InTheHand.Forms MediaElement on iOS. On Android and Windows I can do the following:

```
public partial class MainPage : ContentPage
    {
        public MediaElement mediaView = new MediaElement();
        public MainPage()
        {
            //InitializeComponent();
            var stack = new StackLayout();
            
            mediaView.HeightRequest = 500.0;
            mediaView.WidthRequest = 500.0;
            mediaView.Source = new Uri("https://raw.githubusercontent.com/mediaelement/mediaelement-files/master/big_buck_bunny.mp4");
            mediaView.AreTransportControlsEnabled = true;
            var button = new Button();
            button.Text = "Set Position";
            button.Clicked += Button_Clicked;
            
            stack.Children.Add(mediaView);
            stack.Children.Add(button);

            Content = stack;
        }

        private void Button_Clicked(object sender, EventArgs e)
        {
            var randomNumberGenerator = new System.Random();
            var rand = randomNumberGenerator.NextDouble();
            mediaView.Position = TimeSpan.FromSeconds(10.0+rand); // only works on android and windows not iOS
            //mediaView.Position = TimeSpan.FromSeconds(10.0); //doesn't work on any platform
        }
    }
```

and the position is set as expected. On iOS on the other hand there is a bit of a stutter and then the video resumes from the old position without getting updated.

I have used the trick described here https://github.com/inthehand/InTheHand.Forms/issues/5 to get things working on Android and Windows. If there is an extra trick required for iOS or I am using the package wrong please let me know also!

Thanks!

**Comment by mwpb** [comment]
Please find a minimal example in the following repo:

https://github.com/mwpb/InTheHandSetPosition

**Comment by peterfoot** [comment]
Thanks, reproduced the issue and have been working on a fix. Just need to test for regression on the other platforms.

**Comment by charlesroddie** [comment]
Thanks for fast response Peter!

**Comment by mwpb** [comment]
Hi Peter, thanks for correcting this!

I see that you have updated the git repo but at the moment we are using the NuGet package which appears to be the same (version 1.0.2017.718). Do you think that the NuGet package will be updated soon? Or would it be better for us to try and build from the git repo?

Also it should be possible for me to help test out the set position changes on Windows 8, iOS (8?) and Android 6 if that would be useful to you.

**Comment by peterfoot** [comment]
Thanks, I intended to update the NuGet package this week but have had problems with Visual Studio building the full set of binaries. If you use the git source it uses the new code for position which works much more efficiently. If you can test on those platforms and report back your findings that would be very useful!

**Comment by mwpb** [comment]
Hi, I had a few problems building from the git repo but am now using the updated NuGet package 1.0.2017.804. With this new NuGet package set position seems to work as expected on iOS 10.3 Simulator and Windows 8. Thanks for correcting this. Will test on an iOS 9 device shortly.

However there seems to be a regression on Android: the video doesn't load at all and when I try to load it via a button click I get a null reference exception. In the stack trace I can see some paths that don't correspond to locations on my computer. So I won't update the repo containing my test project just yet. The specific methods are:

InTheHand.Forms.Platform.Android.MediaElementRenderer.UpdateSource
InTheHand.Forms.Platform.Android.MediaElementRenderer.OnElementPropertyChanged

If you would like me to open a different issue please let me know.

**Comment by mwpb** [comment]
Also: if the video hasn't fully loaded on iOS then calling set position causes a crash.

**Comment by peterfoot** [comment]
I will look into these issues (as well as addressing why the code doesn't build directly from the repo) this week

**Comment by mwpb** [comment]
Thanks, to save you some time: I can actually build directly from the repo. It turns out the problem was at my end when I was trying to integrate it with my main project.

About the iOS crash, perhaps it might be convenient to have some way of knowing when the video has loaded? (On Windows 8 and and iOS the buffering progress always seems to give 0 and MediaOpened doesn't fire either.) Then one could add an if clause at the beginning of any method calling set position.

**Comment by peterfoot** [comment]
Android Source issue will be followed up in this separate issue - https://github.com/inthehand/InTheHand.Forms/issues/12

**Comment by peterfoot** [comment]
Fixed in 1.2017.810 iOS will now check the reported valid seek ranges from the native control so will not allow the position to be set if not valid including if the asset is not yet ready to play.

**Comment by mwpb** [comment]
Hi, just to confirm: this appears to be working now in my main project. Thanks!

**Comment by lrolleman** [comment]
This appears to have regressed since it was closed, most recent package (1.0.2017.1107) no longer works when setting the position on Android (and possibly other platforms as well, will check).

810 still works, not sure which release along the way it broke on.

**Comment by SMSteve** [comment]
I had this error, but it turned out to be the NSUrl.FromString on line 153 of iOS.MediaElementRenderer returning NULL.
After integrating the MediaElement code directly into my solution and debugging, I found that by removing any spaces or 'special characters' (like '%20') from the url allowed NSUrl.FromString to process the Element.Source.
Now the Nuget package v1.0.2017.1113 works.
